lib/schemaUtils.js

```diff
@@ -2702,10 +2702,10 @@ module.exports = {
 
                   // for empty string _.set creates new key with empty string '', so separate handling
                   if (dataPath === '') {
-                    suggestedValue = this.getSuggestedValue(fakedValue, valueToUse, ajvError);
+                    suggestedValue = this.getSuggestedValue(fakedValue, suggestedValue, ajvError);
```

Comment:
The tests only test for type, not the actual value. We should test for this change

Reply:
Can you describe this change a little more? What would fakedValue, suggestedValue, valueToUse be for a type mismatch?

Reply:
Ok, so `fakedValue` is value that is faked via json-schema-faker based on schema, `valueToUse` is actual value that is currently present in transaction and will be used as base to which we will override fixes. `suggestedValue` is value that is to be suggested which fixes issue.

Now what happened before was instead retaining changes done in previous iteration (`suggestedValue`), we were always using same `valueToUse` this worked for properties that had one violation, but for property that was violating multiple keyword was only suggested fix that patches one violation.

Reply:
> The tests only test for type, not the actual value. We should test for this change

The test is checking for `name` and `identifier` both properties to be present, which was not happening before.